spec/array_of_integer.rb

```diff
@@ -0,0 +1,26 @@
+require_relative '../part1/array_of_integer'
+require 'pry'
+
+RSpec.describe 'array_of_integer?' do
+  subject { [] }
+
+  context 'empty array' do
+    it 'returns true' do
+      expect(array_of_integer?(subject)).to eq true
+    end
+  end
+
+  context 'all items are integer' do
+    before { subject.push(1,2,4) }
+    it 'returns true' do
+      expect(array_of_integer?(subject)).to eq true
+    end
+  end
+
+  context 'items are of different type' do
+    before { subject.push(:a, "c", 1) }
```

Comment:
bettter not to use subject fot such cases

and you simple let

let(array) {[:a, "c", 1]}